docker/puppetserver/Dockerfile

```diff
@@ -24,6 +24,7 @@ ENV PUPPERWARE_ANALYTICS_TRACKING_ID="UA-132486246-4" \
     PUPPETSERVER_JAVA_ARGS="-Xms512m -Xmx512m" \
     PATH=/opt/puppetlabs/server/bin:/opt/puppetlabs/puppet/bin:/opt/puppetlabs/bin:$PATH \
     SSLDIR=/etc/puppetlabs/puppet/ssl \
+    CADIR=/etc/puppetlabs/puppetserver/ca \
```

Comment:
Do we expect users to change this here?

If not, we've been trying to move `ENV` vars out of the `Dockerfile` because it signals something that can / should be touched by end users.  Instead, better to just embed it in the script where it's used (in this case maybe do a `puppet config print` to retrieve it?)

Puppetserver is the one container that's a bit out of date with the "best practices" we've been trying to put in everywhere -- partially because its the first container, and partially b/c there's a lot of practices to unwind that people may be relying on b/c we have a lot of active users.